hivemind/client/averaging/__init__.py

```diff
@@ -196,14 +197,14 @@ def run_in_background(self, await_ready=True, timeout=None):
     def shutdown(self) -> None:
         """ Shut down the averager process """
         # TODO notify peers before terminating
-        if self.is_alive():
+        if self._parent_pid != os.getpid() or self.is_alive():
             self._pipe.send(('_SHUTDOWN', None))
             self.terminate()
         else:
             logger.warning("DHT shutdown has no effect: the process is not alive")
 
     def __del__(self):
-        if self.is_alive():
+        if self._parent_pid != os.getpid() or self.is_alive():
```

Comment:
This check fixes an edge case where deleting muiltiple averagers at once may trigger AssertionError "Can only check is_alive for child processes"